wizard.py

```diff
@@ -1044,8 +1044,8 @@ def download_layout():
   layout_metadata = in_toto.models.metadata.Metablock(signed=layout)
 
   # Dump layout to memory file and server to user
-  layout_fp = StringIO.StringIO()
-  layout_fp.write("{}".format(layout_metadata))
+  layout_fp = io.BytesIO()
+  layout_fp.write(("{}".format(layout_metadata)).encode())
```

Comment:
Could you explain this bit here? Is that a py2 -> py3 change, to encode?

Reply:
Send_file will send blank files when using io.StringIO in python 3. 
Someone also opens an issue about this in Flask repository, and one of the members of Flask reply:

> The fact that we support BytesIO is not an indication that we should support StringIO. send_file only expects to be given binary file-like objects. The same issue happens when passing an open file in text mode. The function doesn't deal with encoding, so it wouldn't know how to encode the string to bytes, and I think that should remain up to the user code calling it.

Here is the [link ](https://github.com/pallets/flask/issues/3358) for this issue.

Reply:
Thanks for the background information. For future reference, I find it very useful to include information like that (to a reasonable extent) in the commit message. :)

Reply:
I see. I have updated the commit messages to include more information now.